tests/model/test_fuzzy_logic_model.py

```diff
@@ -1,93 +1,97 @@
 import numpy as np
+import os
+import tempfile
 
-from h1st.model.fuzzy_logic_model import FuzzyLogicModel
+from h1st.model.fuzzy_logic import FuzzyLogicModeler, \
+    FuzzyLogicModel, FuzzyLogicRules
 
 
-class SensorFuzzyLogicModel(FuzzyLogicModel):
-    def add_variables(self):
-        """
-        Add fuzzy variables with membership functions
-        """
-        self.add_variable(
-            range_=np.arange(0, 10, 0.5),
-            name='sensor1',
-            membership_funcs=[('normal', 'gaussian', [3, 3.3]),
-                              ('abnormal', 'triangle', [8, 15, 15])],
-            type_='antecedent'
-        )
-        self.add_variable(
-            range_=np.arange(0, 10, 0.5),
-            name='sensor2',
-            membership_funcs=[('normal', 'gaussian', [3, 3.3]),
-                              ('abnormal', 'triangle', [8, 15, 15])],
-            type_='antecedent'
-        )
-        self.add_variable(
-            range_=np.arange(0, 10, 0.5),
-            name='problem1',
-            membership_funcs=[('no', 'trapezoid', [0, 0, 4, 6]),
-                              ('yes', 'trapezoid', [4, 6, 10, 10])],
-            type_='consequent'
-        )
+def build_fuzzy_logic_model():
+    flr = FuzzyLogicRules()
+    flr.add_variable(
+        range_=np.arange(0, 10, 0.5),
+        name='sensor1',
+        membership_funcs=[('normal', 'gaussian', [3, 3.3]),
+                            ('abnormal', 'triangle', [8, 15, 15])],
+        type_='antecedent'
```

Comment:
Shouldn’t the name and range of a variable be *separate* from its membership functions? The same variable may be `hot` for a range of temperatures in one context, and `hot` for a different range of temperatures in a different context.

Reply:
@ctn Could you give me an example so that I can understand the problem better ?

Reply:
Suppose you have `sensor1` measuring temperature. In one use case, `hot` for that `sensor1` may be between 15℃ and 20℃. But in another use case, `hot` may be above 30℃. To support this, you would need to have a one-to-many mappings between `name` and `membership_funcs`.

However, for now, let’s just keep it simple, the way you have done it.